docs/faq/faq.md

```diff
@@ -274,3 +274,38 @@ If you don't find the issue you could also ask for help in our public
 [Element](https://riot.im/app/#/room/#ink:matrix.parity.io) or 
 [Discord](https://discord.gg/ztCASQE) channel.
 
+
+### What are the `scale::Encode` and `scale::Decode` trait?
+
+Substrate-based blockchains use the [SCALE codec](https://www.substrate.io/kb/advanced/codec)
+to encode data.
+As a consequence the data for every interaction with Substrate needs to
+be SCALE-encodable ‒ i.e. it needs to implement either `scale::Encode`,
+`scale::Decode`, or both. This affects e.g. data you want to return to a caller,
+data that you want to take as input, or data you want to store on-chain.
+
+A common error you might get when a necessary SCALE trait is not implemented
+for a data structure could be along the lines of `the trait "WrapperTypeEncode"
+is not implemented for "Foo"`.
```

Comment:
This is also a common (and confusing) error when there are mismatching versions of `parity-scale-codec`

Reply:
Argh.

Reply:
@ascjones Mismatching versions in the contract and as a dependency of ink! ? Could we detect this?

Reply:
Yes exactly. I suppose it would be possible to detect that in `cargo-contract`.

Reply:
I updated this PR and created a ticket for it.